### PR TITLE
You can choose inputs to Google search

### PR DESCRIPTION
**Pull request by mariusjohan**
I've changed it so you don't have to spend time waiting for something you don't really have to use and thereby save some time.

### CODE REVIEW

**Comment by schlichtkrull** [comment]
I also recommend you can choose the amount of results you get since I think the program feels pretty slow.

**Comment by MeNsaaH** [comment]
@schlichtkrull If i understand your PR correctly, you want to return just a single result from the google search?

**Comment by schlichtkrull** [comment]
@MeNsaaH Just to do the program more flexible and automated so the user can those what parametes he/she wants and of course faster.

**Comment by MeNsaaH** [comment]
I understand what you want to do. If a User wants to retrieve just links it is not necessary to get all the full (links, descriptions and title) before returning just the links. 

But this PR would also involve refactoring part of the CLI and other Engines as well.

File: search_engine_parser/core/engines/google.py [hunk 1]
```diff
@@ -33,21 +35,29 @@ def parse_single_result(self, single_result):
...
+        
+	# Get the text and link
+	if 'h3_tag' in vars:
+            title = h3_tag.text
+            if not title:
+                title = h3_tag.find('div', class_='ellip').text
+	    results['titles'] = title
+	
+        if 'link_tag' in vars:
+            raw_link = link_tag.get('href')
```
Comment (MeNsaaH) [inline search_engine_parser/core/engines/google.py]: `link_tag` is not defined here

File: search_engine_parser/core/engines/google.py [hunk 2]
```diff
@@ -24,7 +24,9 @@ def parse_soup(self, soup):
         # find all class_='g' => each result
         return soup.find_all('div', class_='g')
 
-    def parse_single_result(self, single_result):
+    # Elements is a 2d array consting of names, html tags and classes
+    # Elements = [['link_tags', 'a', 'r'], ['r_elem', 'div', 'st'] ... ]
+    def parse_single_result(self, single_result, element):
```
Comment (MeNsaaH) [inline search_engine_parser/core/engines/google.py]: the `element` args should be changed to a keyword `mode` whose default is `full`. Using this `mode`, you can then choose to return process appropriately. This reduces the complexities of having to pass element definition everytime this function should be called. All logic about an engine should be in the engine and not passed as extra args.

File: search_engine_parser/core/engines/google.py [hunk 3]
```diff
@@ -33,21 +35,29 @@ def parse_single_result(self, single_result):
...
+            if not title:
+                title = h3_tag.find('div', class_='ellip').text
+	    results['titles'] = title
+	
+        if 'link_tag' in vars:
+            raw_link = link_tag.get('href')
+	    results['links'] = raw_link
+
+	if 'desc' in vars:
+            desc = desc.text
```
Comment (MeNsaaH) [inline search_engine_parser/core/engines/google.py]: `desc` is also not defined. 

Consider using something like:

```python
results = {}
# This should be replicated for links and titles as well
if mode=="full" or mode == "descriptions":
	desc = single_result.find('span', class_='st')
	results["desc"] = desc.text
```

File: search_engine_parser/core/engines/google.py (hunk 1 above)
Comment (MeNsaaH) [reply search_engine_parser/core/engines/google.py]: Check the last review for how to go about this

**Comment by MeNsaaH** [comment]
Check the current [googlescholar.py](https://github.com/bisoncorps/search-engine-parser/blob/3bab38d0723d2f04eaa5c0f13890c7a33b6ac88b/search_engine_parser/core/engines/googlescholar.py#L26) for a guide to proceed on this

**Comment by MeNsaaH** [comment]
@all-contributors add @mariusjohan  for code

**Pull request merged**